blocks/author-profiles/author-profiles.js

```diff
@@ -0,0 +1,44 @@
+import { loadCSS } from '../../scripts/aem.js';
+import ffetch from '../../scripts/ffetch.js';
+import { div } from '../../scripts/dom-builder.js';
+import { completeEntry, renderProfile } from '../author-profile/author-profile.js';
+
+async function getAuthorEntries(keys) {
+  const entryFilter = ((entry) => (keys.includes(entry.path)));
+  const unsortedEntries = await ffetch('/authors-index.json').filter(entryFilter).limit(keys.length).all();
+  const sortedEntries = [];
+  keys.forEach((key) => {
+    // eslint-disable-next-line max-len
+    sortedEntries.push(completeEntry(unsortedEntries.find((entry) => (key === entry.path))));
+  });
+  return sortedEntries;
+}
+
+async function addAuthorProfiles(block, keys) {
+  const entries = await getAuthorEntries(keys);
+  if (keys.length > 1) {
+    block.classList.add(`elems${keys.length}`);
+    entries.forEach((entry) => {
+      const profile = div({ class: 'author-profile hor' }, renderProfile(entry));
+      block.append(profile);
+    });
+  } else {
+    block.classList.add('vertical');
+    block.append(div({ class: 'author-profile' }, renderProfile(entries[0])));
+  }
+}
+
+export default async function decorateBlock(block) {
+  loadCSS(`${window.hlx.codeBasePath}/blocks/author-profile/author-profile.css`);
+  const keys = [];
+  block.querySelectorAll(':scope p').forEach((para) => {
+    const link = para.querySelector('a');
+    keys.push(link ? (new URL(link.href).pathname) : para.innerText);
+  });
+  if (keys.length === 0) {
+    const link = block.querySelector('a');
+    keys.push(link ? (new URL(link.href).pathname) : block.innerText);
+  }
```

Comment:
Why not directly `block.querySelectorAll('a')` like

```
block.querySelectorAll('a').forEach((link) => {
    keys.push(link.pathname : link.textContent);
  });
```

Reply:
Good catch. This was a left-over from the time when we supported both links and names